src/theory/theory_model.h

```diff
@@ -98,29 +104,39 @@ class TheoryModel : public Model
   void addSubstitution(TNode x, TNode t, bool invalidateCache = true);
   /** assert equality holds in the model
    *
-   * This method returns true if the equality engine of this model is consistent
-   * after asserting the equality to this model.
+   * This method returns true if and only if the equality engine of this model
+   * is consistent after asserting the equality to this model.
    */
   bool assertEquality(TNode a, TNode b, bool polarity);
   /** assert predicate holds in the model
    *
-   * This method returns true if the equality engine of this model is consistent
-   * after asserting the predicate to this model.
+   * This method returns true if and only if the equality engine of this model
+   * is consistent after asserting the predicate to this model.
    */
   bool assertPredicate(TNode a, bool polarity);
   /** assert all equalities/predicates in equality engine hold in the model
    *
-   * This method returns true if the equality engine of this model is consistent
-   * after asserting the equality engine to this model.
+   * This method returns true if and only if the equality engine of this model
+   * is consistent after asserting the equality engine to this model.
    */
   bool assertEqualityEngine(const eq::EqualityEngine* ee,
                             std::set<Node>* termSet = NULL);
   /** assert representative
-    *  This function tells the model that n should be the representative of its equivalence class.
-    *  It should be called during model generation, before final representatives are chosen.  In the
-    *  case of TheoryEngineModelBuilder, it should be called during Theory's collectModelInfo( ... )
-    *  functions.
-    */
+   *
+   * This function tells the model that all terms in the equivalence class of n
```

Comment:
This function gives a "skeleton" for the model value of the equivalence class containing n.  This should be an interpreted function (e.g. datatype constructor, array store) applied to variables that will be filled in with model values from other theories.